src/Foundatio.Repositories.Elasticsearch/Configuration/DailyIndex.cs

```diff
@@ -131,13 +135,19 @@ protected override DateTime GetIndexDate(string index)
         return DateTime.MaxValue;
     }
 
-    private readonly Dictionary<DateTime, object> _ensuredDates = new();
     protected async Task EnsureDateIndexAsync(DateTime utcDate)
     {
         utcDate = utcDate.Date;
         if (_ensuredDates.ContainsKey(utcDate))
             return;
 
+        await using var indexLock = await _ensureIndexLock.AcquireAsync($"Index:{GetVersionedIndex(utcDate)}", TimeSpan.FromMinutes(1)).AnyContext();
+        if (indexLock is null)
+            throw new Exception("Unable to acquire index lock");
```

Comment:
We should check to see if another process created the index we are looking for instead of just throwing here.